client/rpmtrans.c

```diff
@@ -480,9 +465,51 @@ TDNFTransAddInstallPkg(
                   fp,
                   pszFilePath,
                   &rpmHeader);
-    //If not checking gpg sigs, ignore signature errors
-    if(!nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
+
+    Fclose(fp);
+    fp = NULL;
+
+    if(nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
```

Comment:
Please check for negative condition first, that saves a level of indentation and no need of else block